[WIP] New Midround Antagonist: Melting Abnormality!

tralezab opened the pull request:
## "Again?"
Sorry, the last version was not polished and complete. In theory, it could have been testmerged, but it was still a pain for maintainers and it would not have shown off the antagonist's best foot forward. The rest of this is mostly just from the last pr, since I didn't want to rewrite stuff.

The mastermind behind this is @FuryMcFlurry, who made all of the sprites and design. I just coded it all and put it in a pr. Hell, most of this pr is written by him too!

## About The Pull Request

Adds a new mid round antagonist, D-03-109 (better known as Melting Love). It spawns in the maintenance areas of the station.

Melting Abnormality has two abilities, mark and slime.

### Mark
Mark can target a single person to become their “Champion”
The marked champion will still be human but will join sides with the Abnormality, being able to communicate with each other silently.
The champion will get their heart replaced by a new one, which heals them overtime.
Should the champion die, instead their body would melt and increase in size turning them into a greater slime.

The champion periodically spreads an airborne disease that can be counteracted with a gasmask. The champion himself is immune to the effects of this disease.

### Slimeball

Just a ranged projectile on a very high cooldown that infects and stuns the target. This is the only self defense the abnormality gets, so be careful!

### The disease

The disease is globally cured when the Abnormality dies
If the infected person’s body temperature is below its normal heat, the disease’s effects are halted.
Also the medical HUD’s shows an unique icon when someone is infected with said disease.

When people reach stage 5 of the disease they will collapse to the ground and undertake a transformation turning them into a lesser slime.

These slimes are resistant to brute damage, fire extinguishers work well at damaging them though.
Beware of their attacks as they infect people with the disease, even if they are wearing a gasmask.


## Why It's Good For The Game

This is very similar to the jungle fever and i'm even considering replacing it as a gamemode. Jungle fever is lacking as a gamemode; It's disease based but the spread of that disease is blocked by items you can find roundstart which you would always put on because it's very easy to notice people in mass turning into monkeys.

By adding a bit more subtlety to the spreading of the disease and the airborne element making the spreading easier, hopefully this midround variant of that idea should work a bit better to threaten the station.

I had something to say about spider invasions not being a thing anymore and this would replace that as a simplemob thing, but I got nothing really

## Changelog
:cl: Fury McFlurry
add: There's a new midround antagonist in rotation.
/:cl:

tralezab committed:
reminder to self: fix the strings and regex

tralezab committed:
testmergable state

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into melting-love

tralezab committed:
progress, errors with yalp elor? is this revenge?

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into melting-love

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into melting-love

tralezab committed:
more general work + sprites

talk bubbles!

tralezab committed:
mask

tralezab committed:
okay really, THE mask

tralezab committed:
more cleanup i wanted to do before pring

tralezab commented:
reposting the pogchamp icons from last pr
![image](https://user-images.githubusercontent.com/40974010/62418095-3e2a3500-b616-11e9-9a7c-b07567f72da5.png)

tralezab committed:
compiles

tralezab committed:
lint things

tralezab committed:
fucking lint things

knacker48 commented:
How does the Mark ability work? Is it like a projectile that you could actually avoid or is it just like a drop down menu and pick someone you see like mindswap?

mrhugo13 commented:
ummm so basically



fetish cuntent

AarontheIdiot commented:
should be called slime lord because turns people into slimes

tralezab commented:
@knacker48 
close range single use ability that converts a single person by replacing their heart with a special one, they could in theory be deconverted by removing the heart but i doubt that will happen often. their role is to spread the disease. they have it, they spread it, but they do not have any of it's effects themselves and their slime disease never goes past the first stage. They'll be spotted as diseased on a medhud.
@mrhugo13 
yeah, probably will change because maintainers aren't even really that interested for more midround antags at the moment and it turning into a shitty farty feedback thread is just annoying
@AarontheIdiot 
i actually really like the randomly generated names for slime boi

ShadowDeath6 commented:
cool pr

Not sure if heart removal kills outright, but I think trying to remove it should just turn the person into the greater slime like dying does. 

also make sure to restrict the slimes from hitting melting love

tralezab commented:
i'll consider it, the lesser slimes drop the brain of the person transformed anyways

and good idea

fludd12 commented:
Excellent as before. I was wondering, though - why make the complex `generated_creatorname()` proc rather than just making `melting_first.txt` have paired descriptors?

For example,
`Alluring:Allured`
`Blissful:Blissed`
`Ceaseless:Ceased`

And then give the abnormality / virus a variable that determines the slime name, having the abnormality's `Initialize` take apart the first name with regex.

The current system already works, so don't feel the need to look into it too much, but I was wondering if you had a reason for doing it the current way, outside of ease of additions in the future?

tralezab commented:
i could move the system over to paired descriptors, i was just learning regex. i also kinda like that the proc only has to run once, but that's just me having fun.

as for putting the infront of the names, i actually like how that looks. i'll do that.

Qustinnus commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -428,7 +428,7 @@
 	return ..()
 
-/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, toxic = FALSE, harm = TRUE, force = FALSE)
+/mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, specialcolor = null, stun = TRUE, distance = 1, message = TRUE, toxic = FALSE, harm = TRUE, force = FALSE)
```
you cant just jam a new param in the middle of a proc, this'll break most of the calls to this proc because the third param is now special_color instea of stun, which means almost every instance of vomit() will be broken.

tralezab replied:
fixed

Qustinnus commented on code/modules/antagonists/melting/melting_spells.dm:
```diff
@@ -0,0 +1,108 @@
+
+/obj/item/projectile/slime/on_hit(mob/living/carbon/target)
+	. = ..()
+	if(ishuman(target))
```
nesting 4noraisin

Qustinnus commented on code/modules/antagonists/melting/melting.dm:
```diff
@@ -0,0 +1,144 @@
+
+/mob/living/simple_animal/hostile/melting/proc/setup_icons(decided_color)
+	cut_overlays()
+	slimebody_color = rgb(rand(100, 255), rand(100, 255), rand(100, 255))
```
why do this if you might potentially override it 2 lines down?

IndieanaJones commented:
> maintainers aren't even really that interested for more midround antags at the moment
Is this true?  Wouldn't adding more midround antags add more variety to the game, which is what we want?

tralezab committed:
fixes horrible issue that would have arose

Akrilla commented:
It's more that it has been mentioned a few times that there might be an antag freeze soon.

tralezab committed:
the rest of qustinnus's review

tralezab committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into melting-love

shellspeed1 commented:
Hmm, I would recommend transforming people who got turned into slimes back into themselves when the melting abnormality is killed, otherwise looks good.

tralezab commented:
all the slimes drop the people's brains so the recovery process is possible

tralezab committed:
see description, more features & fixes

ambience alert, melting greet, cold stops the disease, ticked the event on (oh god fugitives) and fixed a lot of problems with it

TheChosenEvilOne commented:
I would recommend making a midround ruleset for this in dynamic.

tralezab commented:
will do, thanks

tralezab committed:
TODO: remove vomit shit

optimumtact commented:
@tgstation/commit-access closing in 24 hours unless someone merges

optimumtact commented:
err nvm it's not mergable

tralezab commented:
guess tralezab is working overtime, then! if you don't think I can finish this in 24 hours, you haven't been paying attention.

tralezab committed:
communicate spell for melting + champion, all the champion code, some more icons

todo: big slimer, vomit cleanup, placeholder icons out

tralezab committed:
compile

ExcessiveUseOfCobblestone commented:
If any maintainer is interested reopen I guess, I'm personally not really sold on the whole "Side antag with conversion mechanics"

FuryMcFlurry commented:
Who are the other maintainers? And how would I present this to them. I was hoping my time and effort will be included in ss13.

ExcessiveUseOfCobblestone commented:
https://github.com/orgs/tgstation/teams/commit-access/members

ExcessiveUseOfCobblestone commented:
Probably easiest to have a conversation with them on discord.

FuryMcFlurry commented:
Link leads to nothing, and discord doesn't seem to have a maintainer rank(?) to ping.

optimumtact commented:
it's not pingable, and if they didn't merge this PR they don't care
